Dockerfile.dev

```
@@ -0,0 +1,20 @@
+# Dockerfile.dev is a stripped down version of Dockerfile that is used for development purposes.
+# It minimizes number of installs and builds to speed up the development process.
+# Run this on local host only.
+# Generated image is larger than production image. Do not use this for production.
+
+FROM --platform=${BUILDPLATFORM:-amd64} node:22-slim AS build_dev
```

could use node image here which has a lot of tools pre-installed

```suggestion
FROM --platform=${BUILDPLATFORM:-amd64} node:22 AS build_dev
```

this should in theory allow us to skip extra installation step done on Line 10

the trade-off is that the image is larger but since we don't plan to share it externally I am not sure that matters, would need to see myself if it's worth in terms of build times

---

I ran this locally but I don't see any noticeable difference between them (maybe ~5s) because the `apt-get install` is quite fast. But a second saved is a second saved 👍

---

yeah the install of packages is quite fast..thanks for checking, I wonder if you lose that time again because of increased image size when docker is starting the container

we can probably keep as is and use `-slim`

---

@ensi321 acutally jsut saw here ` => [build_src 3/6] RUN apt-get update && apt-get install -y g++ make python3 python3-setuptools && apt-get cl  36.0s` it says 36 seconds, using the `node` vs. the `node-slim` image makes only a difference on the second build as image will be cached locally

We can get bigger wins though by skipping install stage of node_modules, so makes more sense to focus on that first